scan/git.go

```diff
@@ -151,10 +151,15 @@ func checkoutGit(root, ref, subdir string) (string, error) {
 	// Try checking out by ref name first. This will work on branches and sets
 	// .git/HEAD to the current branch name
 	if output, err := gitWithinDir(root, "checkout", ref); err != nil {
-		// If checking out by branch name fails check out the last fetched ref
-		if _, err2 := gitWithinDir(root, "checkout", "FETCH_HEAD"); err2 != nil {
+		// If the branch name is specified, throw an error
```

Comment:
@xyxyxyxyxyxy I know we never unit test the class before. But I'm wondering if it's possible to unit test it? /cc: @juliusl @shahzzzam

Reply:
This file refers to https://github.com/moby/moby/blob/master/builder/remotecontext/git/gitutils.go a lot.
The moby project has a corresponding test file https://github.com/moby/moby/blob/master/builder/remotecontext/git/gitutils_test.go
Maybe we can refer to that.

Reply:
Unit tests added.